src/main/java/uk/gov/hmcts/ccd/data/user/DefaultUserRepository.java

```diff
@@ -128,24 +131,26 @@ public IdamUser getUser(String userToken) {
     @Override
     public UserDefault getUserDefaultSettings(final String userId) {
         try {
-            final HttpEntity requestEntity = new HttpEntity(securityUtils.authorizationHeaders());
-            final Map<String, String> queryParams = new HashMap<>();
-            queryParams.put("uid", ApplicationParams.encode(userId.toLowerCase()));
+            final HttpHeaders headers = securityUtils.authorizationHeaders();
+            headers.set(EMAIL_HEADER, userId.toLowerCase());
+            final HttpEntity<RestRequest> requestEntity = new HttpEntity<>(headers);
             final String encodedUrl = UriComponentsBuilder.fromHttpUrl(applicationParams.userDefaultSettingsURL())
-                .buildAndExpand(queryParams).toUriString();
-            return restTemplate.exchange(new URI(encodedUrl), HttpMethod.GET, requestEntity, UserDefault.class)
+                .toUriString();
+
+            UserDefaultCollection userDefaultCollection = restTemplate
+                .exchange(new URI(encodedUrl), HttpMethod.GET, requestEntity, UserDefaultCollection.class)
                 .getBody();
+            if (userDefaultCollection == null || isEmpty(userDefaultCollection.getUserDefaults())) {
+                throw new ResourceNotFoundException("User Profile not exists for userId " + userId);
```

Comment:
return a null object, instead of throwing an exception.

Reply:
Done